src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    int pos = 0;
+    for (short i = 0; i < nbrruns; ++i) {
+      short runStart = getValue(i);
+      short length = getLength(i);
+      if (pos + length >= arrayContainer.getCardinality()) {
+        return false;
```

Comment:
I get your point. It's always interesting to get more insights on testing.
So, do you believe that one should add randomised tests on every feature that is implemented ?